### PR TITLE
feat(wasm): add support for Wasm'd getConfig

### PR DESCRIPTION
**Pull request by jkomyno**
## Description

This PR:
- adds support for a Wasm'd `getConfig` function
- uses Wasm'd `getConfig` function as a default regardless of `cliEngineBinaryType`
- unifies the error handling logic between `getConfigWasm` and `getConfigNodeAPI` in the new `parseConfigError` function. This has helped identifying some inconsistencies in the Wasm output that have been fixed in https://github.com/prisma/prisma-engines/pull/3300.
- adds a new `should raise a Rust panic when given arguments of the wrong type` test in `getConfig.test.ts` to ensure we panic when Rust cannot serialize the string value we provide to `get_config`
- adjusts `incomplete-schemas.test.ts` to reflect the new Wasm'd getConfig output

**Note**: the CI is currently expected to fail, as it's missing the engines from the following PRs:
- [x] https://github.com/prisma/prisma-engines/pull/3256
- [x] https://github.com/prisma/prisma-engines/pull/3300

## TODOs

- [x] change signature to [`get_config`](https://github.com/prisma/prisma-engines/blob/5b64b44d7a143117e3659263d6e8d235bd0be054/prisma-fmt/src/get_config.rs#L23-L46) to return a `Result<String, GetConfigError>`
- [x] fix `incomplete-schemas.test.ts`

### CODE REVIEW

File: packages/cli/src/__tests__/incomplete-schemas.test.ts [hunk 1]
```diff
@@ -365,24 +335,40 @@ describe('[normalized library/binary] incomplete-schemas', () => {
...
       } catch (e) {
         expect(serializeQueryEngineName(stripAnsi(e.message))).toMatchInlineSnapshot(
-          urlIsMissingValidationError('getConfig'),
+          urlIsMissingValidationError('getDmmf'),
         )
       }
     })
   })
 
+  // TODO: OK
```
Comment (jkomyno) [inline packages/cli/src/__tests__/incomplete-schemas.test.ts:371]: This comment here it outdated and can be removed.

File: packages/cli/src/__tests__/incomplete-schemas.test.ts [hunk 2]
```diff
@@ -66,21 +67,23 @@ const envVarNotFoundValidationError = `
   Prisma CLI Version : 0.0.0
   `
 
+// - `Validation Error Count` is missing
+// - The error code is missing (as it's set to null in the given error)
```
Comment (jkomyno) [inline packages/cli/src/__tests__/incomplete-schemas.test.ts:71]: This comments are deprecated and should be removed.

File: packages/internals/src/__tests__/engine-commands/getConfig.test.ts [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
-
 import { getConfig } from '../..'
+import { isRustPanic } from '../../panic'
 
 describe('getConfig', () => {
+  test('should raise a Rust panic when given arguments of the wrong type', async () => {
+    expect.assertions(1)
+
+    try {
+      // @ts-ignore
```
Comment (Jolg42) [inline packages/internals/src/__tests__/engine-commands/getConfig.test.ts:9]: Nit: I think `// @ts-expect-error` is usually better

File: packages/internals/src/engine-commands/getConfig.ts [hunk 4]
```diff
@@ -95,6 +108,88 @@ export async function getConfig(options: GetConfigOptions): Promise<ConfigMetaFo
...
+          type: 'parse-json' as const,
+          reason: 'Unable to parse JSON',
+          error: e as Error,
+        }),
+      ),
+    ),
+  )
+
+  if (E.isRight(configEither)) {
+    debug('config data retrieved without errors in getConfig Wasm')
```
Comment (Jolg42) [inline packages/internals/src/engine-commands/getConfig.ts:154]: This looks noisy, we could remove later

**Review by Jolg42 — APPROVED** [review]
Nice! Let's merge this and see how it goes, we can probably remove node-api and binary getConfig soon after :shipit:

File: packages/internals/src/__tests__/engine-commands/getConfig.test.ts (hunk 3 above)
Comment (jkomyno) [inline packages/internals/src/__tests__/engine-commands/getConfig.test.ts:9]:
```suggestion
      // @ts-expect-error
```

Comment (jkomyno) [reply packages/internals/src/__tests__/engine-commands/getConfig.test.ts]: Fair enough, done!

**Pull request merged**